napari/plugins/tests/test_hookspecs.py

```diff
@@ -0,0 +1,87 @@
+import inspect
+import warnings
+
+import pytest
+from numpydoc.docscrape import FunctionDoc
+
+from napari.plugins import hookspecs
+
+# 1. we first create a hookspec decorator:
+#    ``hookspec = pluggy.HookspecMarker("napari")```
+# 2. when it decorates a function, that function object gets a new attribute
+#    called "napari_spec"
+# 3. that attribute is what makes them discoverable when you run
+#    ```plugin_manager.add_hookspecs(module)```
+#
+# here, we are using that attribute to discover all of our internal hookspecs
+# (in module ``napari.plugins.hookspecs``) so as to make sure that they conform
+# to our own internal rules about documentation and type annotations, etc...
```

Comment:
Comment as above — the comment should be in plain, non-jargony English as much as possible!

Reply:
modified: I removed even the word `hookspec` everywhere possible.  The only remaining usages cannot be changed since they refer to methods in pluggy itself.